internal/resolvers/seventv/model.go

```diff
@@ -1,41 +1,62 @@
 package seventv
 
-type EmoteAPIUser struct {
-	ID          string `json:"id"`
-	DisplayName string `json:"display_name"`
-}
+type TooltipData struct {
+	Code     string
+	Type     string
+	Uploader string
 
-type EmoteAPIEmote struct {
-	ID         string       `json:"id"`
-	Name       string       `json:"name"`
-	Visibility int32        `json:"visibility"`
-	Owner      EmoteAPIUser `json:"owner"`
+	Unlisted bool
 }
 
-type EmoteAPIResponseData struct {
-	Emote *EmoteAPIEmote `json:"emote,omitempty"`
+// Definitions from:
+// * Emotes: https://github.com/SevenTV/API/blob/a907ccc44e7eb5bdba7b7e63d2b4b67e0c04f778/data/model/emote.model.go
+// * Users: https://github.com/SevenTV/API/blob/a907ccc44e7eb5bdba7b7e63d2b4b67e0c04f778/data/model/user.model.go
+// * Images: https://github.com/SevenTV/API/blob/a907ccc44e7eb5bdba7b7e63d2b4b67e0c04f778/data/model/model.go
+
+type EmoteModel struct {
+	ID     string           `json:"id"`
+	Name   string           `json:"name"`
+	Flags  EmoteFlagsModel  `json:"flags"`
+	Listed bool             `json:"listed"`
+	Owner  UserPartialModel `json:"owner,omitempty" extensions:"x-omitempty"`
+	Host   ImageHost        `json:"host"`
 }
 
-type EmoteAPIResponse struct {
-	Data EmoteAPIResponseData `json:"data"`
-}
+type EmoteFlagsModel int32
 
-type TooltipData struct {
-	Code     string
-	Type     string
-	Uploader string
+const (
+	EmoteFlagsPrivate   EmoteFlagsModel = 1 << 0 // The emote is private and can only be accessed by its owner, editors and moderators
+	EmoteFlagsAuthentic EmoteFlagsModel = 1 << 1 // The emote was verified to be an original creation by the uploader
+	EmoteFlagsZeroWidth EmoteFlagsModel = 1 << 8 // The emote is recommended to be enabled as Zero-Width
 
-	Unlisted bool
+	// Content Flags
+
+	EmoteFlagsContentSexual           EmoteFlagsModel = 1 << 16 // Sexually Suggesive
+	EmoteFlagsContentEpilepsy         EmoteFlagsModel = 1 << 17 // Rapid flashing
+	EmoteFlagsContentEdgy             EmoteFlagsModel = 1 << 18 // Edgy or distasteful, may be offensive to some users
+	EmoteFlagsContentTwitchDisallowed EmoteFlagsModel = 1 << 24 // Not allowed specifically on the Twitch platform
+)
```

Comment:
Move the comments to be above the definition instead of on the same line - leave an empty line inbetween the definitions

```go
// xd
EmoteFlagXd = 5

// lol
EmoteFlagLol = 6
```